### PR TITLE
(PUP-9071) Fix portage package provider to support versioning

### PR DESCRIPTION
**Pull request by derdanne**
Use package versions and releases in queries and emerge commands

### CODE REVIEW

**Comment by ekinanp** [comment]
Thank you for submitting this PR @derdanne! Can you please edit the commit message to conform to the following format:
```
(<ticket_number>) <summary>

<description>
```

Note that the `(<ticket_number>) <summary>` part should be 80 characters or less. The `<description>` should describe the issue that the PR's fixing, including a brief summary of what the added code here's doing (this would be especially useful for me as I am not at all familiar with the portage package provider [and maybe other reviewers too]). See https://github.com/puppetlabs/puppet/commit/6cb018b1b10e6f5a039453abcd5ad53654b0eff5 as an example.

Thanks again!

**Comment by derdanne** [comment]
I have updated the commit message

File: lib/puppet/provider/package/portage.rb [hunk 1]
```diff
@@ -111,7 +111,14 @@ def qatom
     @atom ||= begin
       package_info = {}
       # do the search
-      search_output = qatom_bin(*([@resource[:name], '--format', output_format]))
+      should = @resource[:ensure]
+      case should
+      when true, false, Symbol
```
Comment (branan) [inline lib/puppet/provider/package/portage.rb]: I *think* what this is doing is relying on `present`, `absent`, and `latest` becoming symbols, while versions are going to come through as strings?

Could you add a comment describing that so someone not super familiar with Puppet and Ruby can understand what's going on here? It's a big magical.

Comment (branan) [reply lib/puppet/provider/package/portage.rb]: It also may be clearer to invert the logic here - `when String` for when we have a version, and the else block can handle everything else

Comment (derdanne) [reply lib/puppet/provider/package/portage.rb]: Hi, i will complete these tasks during next week.

Comment (derdanne) [reply lib/puppet/provider/package/portage.rb]: Hi, i've put a comment in front of this code. But i would keep the logic here consistent with the other package providers like apt and yum, where we use the same logic.

**Comment by jtappa** [comment]
Could someone from @puppetlabs/platform-os address this community PR?

**Pull request merged**